### PR TITLE
audio: fix igo_nr wrapper code issues.

### PR DESCRIPTION
**Pull request by fuyuntsuo**
This PR resolved the issues in #4017 and added license file.
1. Removed NULL assignment of cd `(L223: igo_nr.c)`
2. Removed ENUM in `igo_nr_cmd_get_value` and `igo_nr_cmd_set_value`
3. Added new line in L556:igo_nr.c
4. Moved `struct IGO_PARAMS` from `igo_nr.c` to `igo_nr.h` since `sof_igo_nr_config` needs it.
5. Added license file.

Signed-off-by: fy.tsuo <fy.tsuo@intelli-go.com>

### CODE REVIEW

File: src/audio/igo_nr/igo_nr.c [hunk 1]
```diff
@@ -29,25 +29,6 @@
 
 #define SOF_IGO_NR_MAX_SIZE 4096		/* Max size for coef data in bytes */
```
Comment (dbaluta) [inline src/audio/igo_nr/igo_nr.c:30]: Please add one patch per type of issues fixed.

Comment (fuyuntsuo) [reply src/audio/igo_nr/igo_nr.c:30]: Sure. Just split it to four commits.

**Review by dbaluta — APPROVED** [review]

File: src/audio/igo_nr/igo_nr.c [hunk 2]
```diff
@@ -29,25 +29,6 @@
...
 
-struct IGO_PARAMS {
```
Comment (cujomalainey) [inline src/audio/igo_nr/igo_nr.c:32]: why does this need to be moved?

File: src/audio/igo_nr/igo_nr.c [hunk 3]
```diff
@@ -553,8 +553,8 @@ static void igo_nr_get_igo_params(struct comp_dev *dev)
 	struct comp_data *cd = comp_get_drvdata(dev);
 
 	struct sof_igo_nr_config *p_config = comp_get_data_blob(cd->model_handler, NULL, NULL);
-	comp_info(dev, "igo_nr_get_igo_params()");
 
+	comp_info(dev, "igo_nr_get_igo_params()");
```
[the remark this answers is not in the document]
Comment (fuyuntsuo) [reply src/audio/igo_nr/igo_nr.c:556]: sorry. I just removed that newline.

File: src/audio/igo_nr/igo_nr.c (hunk 2 above)
Comment (fuyuntsuo) [reply src/audio/igo_nr/igo_nr.c:32]: thanks!

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**